src/DocumentFormat.OpenXml/Framework/Schema/ParticleCollection.cs

```diff
@@ -0,0 +1,290 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Validation.Schema;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace DocumentFormat.OpenXml.Framework.Schema
+{
+    internal readonly struct ParticleCollection<TElement> : IEnumerable<TElement>
```

Comment:
This represents a collection of particles at a particular path, right? I found this class a bit confusing since I'm not very familiar with particles. It may be good to put some class/method-level comments explaining expectations here.

Reply:
Yeah, makes sense. Added a few comments.